web/src/main/java/com/bazaarvoice/emodb/web/resources/databus/DatabusResource1.java

```diff
@@ -123,7 +124,13 @@ public SuccessResponse subscribe(@PathParam ("subscription") String subscription
             tableFilter = new ConditionParam(conditionString).get();
         }
 
-        _databus.subscribe(subject, subscription, tableFilter, subscriptionTtl.get(), eventTtl.get(), includeDefaultJoinFilter);
+        Duration subscriptionTtlDuration = subscriptionTtl.get();
+        checkArgument(subscriptionTtlDuration.compareTo(Duration.ofDays(365 * 3)) < 0, "Subscription TTL duration limit is 3 years. The value cannot go beyond that.");
+
+        Duration eventTtlDuration = eventTtl.get();
+        checkArgument(eventTtlDuration.compareTo(Duration.ofDays(365)) < 0, "Event TTL duration limit is 365 days. The value cannot go beyond that.");
```

Comment:
also, here and above extract duration into constant

Reply:
👍